src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseLoader.cs

```diff
@@ -128,6 +196,96 @@ public sealed class Column
+            /// A range representing a set of values. Will result in a vector column.
+            /// </summary>
+            /// <param name="min">The minimum inclusive index of the column.</param>
+            /// <param name="max">The maximum-inclusive index of the column. If <c>null</c>
+            /// indicates that the <see cref="TextLoader"/> should auto-detect the legnth
+            /// of the lines, and read untill the end.</param>
+            public Range(int min, int? max)
+            {
+                Contracts.CheckParam(min >= 0, nameof(min), "Must be non-negative");
+                Contracts.CheckParam(!(max < min), nameof(max), "If specified, must be greater than or equal to " + nameof(min));
+
+                Min = min;
+                Max = max;
+                // Note that without the following being set, in the case where there is a single range
+                // where Min == Max, the result will not be a vector valued but a scalar column.
+                ForceVector = true;
+                AutoEnd = max == null;
+            }
+
+            /// <summary>
+            ///  The minimum index of the column, inclusive.
+            /// </summary>
+            [Argument(ArgumentType.Required, HelpText = "First index in the range")]
+            public int Min;
+
+            /// <summary>
+            /// The maximum index of the column, inclusive. If <see langword="null"/>
+            /// indicates that the <see cref="TextLoader"/> should auto-detect the legnth
+            /// of the lines, and read untill the end.
+            /// If max is specified, the fields <see cref="AutoEnd"/> and <see cref="VariableEnd"/> are ignored.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Last index in the range")]
+            public int? Max;
+
+            /// <summary>
+            /// Whether this range extends to the end of the line, but should be a fixed number of items.
+            /// If <see cref="Max"/> is specified, the fields <see cref="AutoEnd"/> and <see cref="VariableEnd"/> are ignored.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "This range extends to the end of the line, but should be a fixed number of items",
+                ShortName = "auto")]
+            public bool AutoEnd;
+
+            /// <summary>
+            /// Whether this range extends to the end of the line, which can vary from line to line.
+            /// If <see cref="Max"/> is specified, the fields <see cref="AutoEnd"/> and <see cref="VariableEnd"/> are ignored.
+            /// If <see cref="AutoEnd"/> is <see langword="true"/>, then <see cref="VariableEnd"/> is ignored.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "This range extends to the end of the line, which can vary from line to line",
+                ShortName = "var")]
+            public bool VariableEnd;
```

Comment:
I think we should probably leave these options off until we are sure we will need them (or actually use them). I'm not certain `VariableEnd` makes sense for a DatabaseLoader.

Reply:
Removed.